### PR TITLE
Update file expiry with RAG chat

### PR DESCRIPTION
**Pull request by rachaelcodes**
## Context

File expiry dates are set on file creation (since #425 ), but should be updated to give another 30 days before expiry whenever they are quoted in a RAG chat.

## Changes proposed in this pull request

On a call to `/post-message`, the returned source files each have their expiry date extended for another 30 days (or as long into the future as configured with env vars).

## Guidance to review

Could use Django shell to inspect a file expiry date before and after it is used in a RAG chat.

## Relevant links
https://technologyprogramme.atlassian.net/browse/REDBOX-234

## Things to check

- ~[ ] I have added any new ENV vars in all deployed environments~
- [x] I have tested any code added or changed
- ~[ ] I have run integration tests~

### CODE REVIEW

File: django_app/redbox_app/redbox_core/views.py [hunk 1]
```diff
@@ -222,6 +224,10 @@ def post_message(request: HttpRequest) -> HttpResponse:
     files: list[File] = File.objects.filter(core_file_uuid__in=doc_uuids, user=request.user)
     llm_message.source_files.set(files)
 
+    for file in files:
```
Comment (252afh) [inline django_app/redbox_app/redbox_core/views.py:227]: I _think_, according to core, this is just all files for that user, we don't need to use a response from core for that

File: django_app/tests/test_views.py [hunk 2]
```diff
@@ -264,6 +270,10 @@ def test_post_message_to_existing_session(chat_history: ChatHistory, client: Cli
     assert (
         ChatMessage.objects.get(chat_history__id=session_id, role=ChatRoleEnum.ai).text == "Good afternoon, Mr. Amor."
     )
+    assert (
+        ChatMessage.objects.get(chat_history__id=session_id, role=ChatRoleEnum.ai).source_files.first() == uploaded_file
+    )
+    assert initial_file_expiry_date != File.objects.get(core_file_uuid=uploaded_file.core_file_uuid).expiry_date
```
[the remark this answers is not in the document]
Comment (rachaelcodes) [reply django_app/tests/test_views.py:276]: Yeah, good idea :)

File: django_app/redbox_app/redbox_core/views.py (hunk 1 above)
Comment (rachaelcodes) [reply django_app/redbox_app/redbox_core/views.py:227]: Is it not the ones returned as relevant by the LLM chain `core_api/src/routes/chat.py#L150-L151`?

Comment (brunns) [reply django_app/redbox_app/redbox_core/views.py:227]: I believe you're right, Rachael - it's just the "sources", i.e. the relevant files.

File: django_app/redbox_app/redbox_core/models.py [hunk 3]
```diff
@@ -66,18 +66,18 @@ class File(UUIDPrimaryKeyBase, TimeStampedModel):
...
     core_file_uuid = models.UUIDField(null=True)
-    expiry_date = models.DateTimeField(blank=True, null=True)
+    last_referenced = models.DateTimeField(blank=True, null=True)
 
     def __str__(self) -> str:  # pragma: no cover
         return f"{self.original_file_name} {self.user}"
 
     def save(self, *args, **kwargs):
-        if not self.expiry_date:
+        if not self.last_referenced:
```
Comment (252afh) [inline django_app/redbox_app/redbox_core/models.py:75]: I wonder if we need this part if we have the migration? (or vice versa)

**Review by 252afh — APPROVED** [review]

**Pull request merged**